app/controllers/ems_common.rb

```diff
@@ -67,193 +67,34 @@ def show
-      title = "Cloud Tenants"
-      drop_breadcrumb(:name => "#{@ems.name} (All #{title})", :url => show_link(@ems, :display => @display))
-      @view, @pages = get_view(CloudTenant, :parent => @ems) # Get the records (into a view) and the paginator
-      @showtype = @display
-      if @view.extras[:total_count] && @view.extras[:auth_count] && @view.extras[:total_count] > @view.extras[:auth_count]
-        @bottom_msg = "* You are not authorized to view " + pluralize(@view.extras[:total_count] - @view.extras[:auth_count], "other #{title.singularize}") + " on this " + ui_lookup(:tables => @table_name)
-      end
-    elsif @display == "cloud_volumes" || (session[:display] == "cloud_volumes" && params[:display].nil?)
-      title = "Cloud Volumes"
-      drop_breadcrumb(:name => "#{@ems.name} (All #{title})", :url => show_link(@ems, :display => @display))
-      @view, @pages = get_view(CloudVolume, :parent => @ems) # Get the records (into a view) and the paginator
-      @showtype = @display
-      if @view.extras[:total_count] && @view.extras[:auth_count] && @view.extras[:total_count] > @view.extras[:auth_count]
-        @bottom_msg = "* You are not authorized to view " + pluralize(@view.extras[:total_count] - @view.extras[:auth_count], "other #{title.singularize}") + " on this " + ui_lookup(:tables => @table_name)
-      end
-    elsif @display == "flavors" || session[:display] == "flavors" && params[:display].nil?
-      title = "Flavors"
-      drop_breadcrumb(:name => @ems.name + " (All #{title})", :url => show_link(@ems, :display => @display))
-      @view, @pages = get_view(Flavor, :parent => @ems)  # Get the records (into a view) and the paginator
-      @showtype = @display
-      if @view.extras[:total_count] && @view.extras[:auth_count] &&
-         @view.extras[:total_count] > @view.extras[:auth_count]
-        @bottom_msg = "* You are not authorized to view " + pluralize(@view.extras[:total_count] - @view.extras[:auth_count], "other #{title.singularize}") + " on this " + ui_lookup(:tables => @table_name)
-      end
-    elsif @display == "security_groups" || session[:display] == "security_groups" && params[:display].nil?
-      title = "Security Groups"
-      drop_breadcrumb(:name => @ems.name + " (All #{title})", :url => show_link(@ems, :display => @display))
-      @view, @pages = get_view(SecurityGroup, :parent => @ems)  # Get the records (into a view) and the paginator
-      @showtype = @display
-      if @view.extras[:total_count] && @view.extras[:auth_count] &&
-         @view.extras[:total_count] > @view.extras[:auth_count]
-        @bottom_msg = "* You are not authorized to view " + pluralize(@view.extras[:total_count] - @view.extras[:auth_count], "other #{title.singularize}") + " on this " + ui_lookup(:tables => @table_name)
-      end
+    elsif display_name = (Set.new([@display, (session[:display] unless params[:display])]) &
+                          Set.new(%w(containers
```

Comment:
I'd prefer something like
`(display_name = calculate_display_name).present?`
to make it more readable?

and then we could have a simple spec around `calculate_display_name` to capture the possibilities such as `params[:display]` and `session[:display]` present or not present and value (not) being in the set